master/pkg/model/defaults.go

```diff
@@ -29,14 +29,52 @@ const (
 	defaultGPUImage = "determinedai/environments:cuda-10.2-pytorch-1.7-tf-1.15-gpu-baefbf7"
 )
 
+// DefaultResourcesConfig returns the default resources configuration.
+func DefaultResourcesConfig(taskContainerDefaults *TaskContainerDefaultsConfig) ResourcesConfig {
+	config := ResourcesConfig{
+		SlotsPerTrial:  1,
+		Weight:         1,
+		NativeParallel: false,
+	}
+	if taskContainerDefaults == nil {
+		return config
+	}
+
+	config.Devices = taskContainerDefaults.Devices
```

Comment:
this isn't the subject of this pr but this behavior is questionable (multiple containers on the same agent could all get access to all the devices on the cluster, this is really a scheduler deficiency).